### PR TITLE
[agent][prtg] PRTG integration

### PR DESCRIPTION
**Pull request by vpoliukh**
- PRTG source
- PRTG pipeline with parsing .xml or .json responses
- Unittests added

Related ticket: [AL-11312](https://anodot.atlassian.net/browse/AL-11312)

### CODE REVIEW

File: agent/src/agent/pipeline/config/jython_scripts/prtg.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+                        break
+            else:
+                record = sdc.createRecord('record created ' + str(datetime.now()))
+                record.value = {'last_timestamp': int(offset)}
+                cur_batch.add(record)
+
+            cur_batch.process(entityName, str(offset))
+            offset += interval.total_seconds()
+        except requests.HTTPError as e:
+            requests.post(sdc.userParams['MONITORING_URL'] + str(e.response.status_code))
```
[the remark this answers is not in the document]
Comment (vpoliukh) [reply agent/src/agent/pipeline/config/jython_scripts/prtg.py:151]: Yep, missed it. Thanks

File: agent/src/agent/pipeline/config/jython_scripts/prtg.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+def get_now():
+    return int(time.time())
+
+
+def to_timestamp(date):
+    epoch = datetime(1970, 1, 1)
+    return int((date - epoch).total_seconds())
+
+
+def prtg_ts_to_unix_ts(prtg_ts, tz=pytz.timezone(sdc.userParams['TIMEZONE'])):
```
[the remark this answers is not in the document]
Comment (vpoliukh) [reply agent/src/agent/pipeline/config/jython_scripts/prtg.py]: Removed

File: agent/src/agent/pipeline/config/jython_scripts/prtg.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    return datetime.fromtimestamp((prtg_ts - 25569) * 86400)
+
+
+def _filter(list_):
```
[the remark this answers is not in the document]
Comment (vpoliukh) [reply agent/src/agent/pipeline/config/jython_scripts/prtg.py]: Removed

File: agent/src/agent/pipeline/config/jython_scripts/prtg.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    ts_unix = get_now()
+    devices = tree.findall('*//device')
+    for device in devices:
+        device_name = device.find('name')
+        sensors = device.findall('sensor')
+        for sensor in sensors:
+            sensortype = sensor.find('sensortype')
+            name = sensor.find('name')
+            value = sensor.find('lastvalue_raw')
+            prtg_ts = sensor.find('lasttime_raw_utc')
```
[the remark this answers is not in the document]
Comment (vpoliukh) [reply agent/src/agent/pipeline/config/jython_scripts/prtg.py]: Removed

**Review by dariakharlan — APPROVED** [review]

**Pull request merged**